test/suite/test_encrypt05.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_encrypt03.py
+#   Test some error conditions with quoted escaped characters.
+#
+
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+
+# Test error conditions with quoted escaped characters.
+class test_encrypt05(wttest.WiredTigerTestCase):
+
+    types = [
+        ('table', dict(uri='table:test_encrypt05')),
+    ]
+    encrypt = [
+        ('rotn', dict( sys_encrypt='rotn', sys_encrypt_args=',keyid=11',
+            file_encrypt='rotn', file_encrypt_args=',keyid=13')),
+    ]
+    scenarios = make_scenarios(types, encrypt)
+
+    def conn_extensions(self, extlist):
+        extlist.skip_if_missing = True
+        extlist.extension('encryptors', self.sys_encrypt)
+        extlist.extension('encryptors', self.file_encrypt)
+
+    def conn_config(self):
+        return 'encryption=(name={0}{1}),'.format(
+            self.sys_encrypt, self.sys_encrypt_args)
+
+    # Open connection with a config that has a malformed keyid.
+    # The keyid is malformed because it contains a quoted escaped character.
+    def test_encrypt(self):
+        escaped_chars = ['\n', '\r', '\t', '\b']
```

Comment:
I would create scenarios, with escaped_chars as the changing one. e.g:
```
    encrypt = [
        (' check new line rotn', dict( sys_encrypt='rotn', sys_encrypt_args=',keyid=11',
            file_encrypt='rotn', file_encrypt_args=',keyid=13', escaped_char="\n")),
         'check tab spaces', dict( sys_encrypt='rotn', sys_encrypt_args=',keyid=11',
            file_encrypt='rotn', file_encrypt_args=',keyid=13',escaped_char="\t")),
      ]
```

Reply:
Based on our chat I reworked by leveraging the `make_scenarios` call. See [30fc141](https://github.com/wiredtiger/wiredtiger/pull/12194/commits/30fc141bd86410bc378e72dc2a0cda12d0ff98a4)